design/external-references.md

```diff
@@ -0,0 +1,394 @@
+// in any stack in the same account/region
+const importValues = user.export();
+
+// instantiates an `IUser` whose attributes are `Fn::ImportValue` tokens
+// in "stackWithGroup", and it can now be used "naturally" in that stack
+const importedUser = iam.User.import(stackWithGroup, 'ImportedUser', importValues);
+group.addUser(importedUser);
+```
+
+Obvsiouly the world is a better place now. Thanks [@rix0rrr](https://github.com/users/rix0rrr).
+
+## Problem Statement
+
+So, "what seems to be the problem?", you ask. Well, we want more! The current model only supports references that occur between stacks within the same CDK app. What about the use case where I want to reference a resource that was created in a totally different app?
+
+A common example is for a common infrastructure team to maintain a CDK app which defines networking resources such as VPCs, security groups, LBs, etc and application teams need to reference these resources in their own CDK apps.
+
+__So the problem we are trying to solve is how to enable users to reference resources (and potentially higher level constructs) across CDK apps.__
+
+## Scope
+
+As an initial step, we'll address the use case of referencing resources across within __the same AWS environment__ (account + region). This limitation is based on the limitation AWS CloudFormation has for exports/imports.
+
+We differentiate two types of references:
+
+1. Reference a resource created through another CDK app.
+2. Reference a resource created by some other means.
+
+In the former case (app to app), the producing app can explicitly "export" the resource and the consuming app can explicitly "import" the resource.
+
+In the latter case (environment to app), we usually just have a resource's physical identity such as ARN or name and we want to be able to use it in a CDK app.
+
+## Requirements
+
+- **REQ1**: Allow constructs to be published from one CDK app and consumed by other CDK apps within the same environment
+- **REQ2**: Support "simple" constructs, which are resources that can be represented by a set of attributes such as an ARN
+- Support "composite" constructs, which are resources that encapsulate other resources (such as security groups, roles, etc).
+- **REQ3**: Allow CDK apps to reference resources that were created elsewhere (by other CDK apps, manually or any other mean) by referencing the physical identity. For example, an S3 Bucket ARN should be sufficient in order to reference
+an existing S3 bucket.
+- **REQ4**: If possible, implement the mechanism such that it can be later used to publish and consume constructs through other key/value mechanism such as environment variables, SSM parameters, etc.
+- **REQ5**: If a resource is exported twice under the same export name, only a single set of outputs will be created ([#1496](https://github.com/awslabs/aws-cdk/issues/1496)).
+- **REQ6**: If the same resource is imported twice in the same stack, the "import" method will return the same object.
```

Comment:
"same instance". You are right. It's not a nice-to-have requirement. More than that, I actually wanted to say that "it is okay" that it will be the same object. Changed wording a little.